### PR TITLE
45 TC traitor update

### PR DESCRIPTION
**Pull request by Gandalf2k15**
## About The Pull Request

# Base TC now 40

This PR adds 3 items to the traitor uplink along with their respective bundles:

- C20R SMG:
Bundle: 25
Stand alone: 17
Ammo: Standard nukeops.

- Bulldog:
Bundle: 20
Stand alone: 15
Ammo: Standard nukeops.

- Armor Vest
1 TC

- Bulletproof Armor Vest
6 TC

- Syndicate Helmet
4 TC

- Syndicate jaw-o-life
4 TC

- Viscerator grenade
7 TC

- Buzzkill Grenade
15 TC

- Energy shield:
5 TC

- Katana:
12 TC

- CQC
25 TC

The super surplus crate is now 50 TC, and comes with 145 TC of stuff.

Probably more to come.

## Why It's Good For The Game

We buffed security to heaven, we need to bring traitors back upto scratch.

## Changelog
:cl:
add: Traitors now get access to the bulldog, C20R, buzzkill grenade, viscerator grenade, syndie-jaws-o-life, energy shield, armor, helmet, katana, and an alternative revolver.
tweak; Traitors now come with 40 TC as standard.
tweak: Some old stats have been changed regarding uplink items.
/:cl:

### CODE REVIEW

**Comment by Swiftfeather** [comment]
I'd buff the super sized surplus crate that's 40 TC to 50TC and give it 25% more rolls since it's meant to be a "two traitors get together and share their wealth" kind of thing

**Comment by Gandalf2k15** [comment]
> I'd buff the super sized surplus crate that's 40 TC to 50TC and give it 25% more rolls since it's meant to be a "two traitors get together and share their wealth" kind of thing

Agreed

**Comment by JGlitch** [comment]
I'm Johnny Alwaysantagador. And I consider this to be a terrible idea. Please, **PLEASE** do not merge it...

**Comment by BrotherHangyul** [comment]
I'm Hangyul Alwayssecurityim. And I consider this to be a great idea. Please, **PLEASE** do merge it.

File: modular_skyrat/master_files/code/modules/uplink/uplink_items.dm [hunk 1]
```diff
@@ -11,23 +11,77 @@
...
+
+/datum/uplink_item/dangerous/shield_traitor
+	name = "Energy Shield"
+	desc = "An incredibly useful personal shield projector, capable of reflecting energy projectiles and defending \
+			against other attacks. Pair with an Energy Sword for a killer combination."
+	item = /obj/item/shield/energy
+	cost = 5
+	exclude_modes = list(/datum/game_mode/nuclear/clown_ops, /datum/game_mode/nuclear)
+
+/datum/uplink_item/dangerous/shield_traitor
```
Comment (Swiftfeather) [inline modular_skyrat/master_files/code/modules/uplink/uplink_items.dm]: Same as the energy shield path, suggest

```suggestion
/datum/uplink_item/dangerous/katana_traitor
```

**Review by Swiftfeather — CHANGES_REQUESTED** [review]

File: modular_skyrat/master_files/code/modules/uplink/uplink_items.dm [hunk 2]
```diff
@@ -11,23 +11,77 @@
...
+	name = "Katana"
+	desc = "An incredibly sharp sword used by Samurais. Woefully underpowered in D20."
+	item = /obj/item/katana
+	cost = 12
+	exclude_modes = list(/datum/game_mode/nuclear/clown_ops, /datum/game_mode/nuclear)
+
+//BUNDLES
+
+/datum/uplink_item/bundles_tc/bulldog
```
Comment (Swiftfeather) [inline modular_skyrat/master_files/code/modules/uplink/uplink_items.dm:63]: Shares the same path as the normal bulldog bundle.

```suggestion
/datum/uplink_item/bundles_tc/bulldog_traitor
```

**Comment by Gandalf2k15** [comment]
We'll try 40 TC

**Comment by Swiftfeather** [comment]
Hey gandalf, can you change all the guns added to traitors from the nukies to /unrestricted subtype? They come with the syndicate firing pin by default meaning traitors can't shoot them.

**Comment by Gandalf2k15** [comment]
> Hey gandalf, can you change all the guns added to traitors from the nukies to /unrestricted subtype? They come with the syndicate firing pin by default meaning traitors can't shoot them.

Yep!

**Comment by CameronWoof** [comment]
Since you asked for comments, I do not think this is a good idea. Allowing traitors to stockpile even more murder tools is not the way forward. Creating an arms race between antagonists and security is not what we want. What we want is to give traitors access to more tools to frame, hide, and evade. This will just end in more shootouts and more hurt feelings and crying when the antagonist with two deswords, an elite hardsuit and carp gets surrounded and ground into paste by sec batons.

I don't *want* to encourage antagonists to fight and be combative. They have tools for that already. What they need is something that supports any other style of play.

**Comment by Owai-Seek** [comment]
I completely agree with Cameron on this, they don't need more murder tools, they need tools where they can effectively evade security. Right now the best way to evade security is to just.... kill everybody following you, because there's almost nothing when it comes to non-lethal gear for antags.

This is what an antag has to be non lethal, or escape with right now, without TC trades. This isn't including stuff you can be sneaky with, it's actively getting away from security. (This isn't a full list, but I think it nails a decent chunk of stuff.)

- Toy Pistol with Foam Darts
- Mini E-Bow
- Martial Arts Scroll (If just used for dodge.)
- Chameleon Projector
- Stealth Implant
- Freedom Implant (Which if used even a single time, makes you valid.)
- Hypno Flashes
- Stolen Security Gear (Batons, Pepperspray, ect.)
- Sleepy Pen?
- Stimpack?
- Radio Jammer

Most of these options are terrible and only keep sec officers away for a short time, if at all, in an active pursuit. This PR basically just says "Yea, to get rid of security, ya basically gotta kill them."

**Comment by Gandalf2k15** [comment]
> I completely agree with Cameron on this, they don't need more murder tools, they need tools where they can effectively evade security. Right now the best way to evade security is to just.... kill everybody following you, because there's almost nothing when it comes to non-lethal gear for antags.
> 
> This is what an antag has to be non lethal, or escape with right now, without TC trades. This isn't including stuff you can be sneaky with, it's actively getting away from security. (This isn't a full list, but I think it nails a decent chunk of stuff.)
> 
> * Toy Pistol with Foam Darts
> * Mini E-Bow
> * Martial Arts Scroll (If just used for dodge.)
> * Chameleon Projector
> * Stealth Implant
> * Freedom Implant (Which if used even a single time, makes you valid.)
> * Hypno Flashes
> * Stolen Security Gear (Batons, Pepperspray, ect.)
> * Sleepy Pen?
> * Stimpack?
> * Radio Jammer
> 
> Most of these options are terrible and only keep sec officers away for a short time, if at all, in an active pursuit. This PR basically just says "Yea, to get rid of security, ya basically gotta kill them."

I agree with what you're saying here, I'd appreciate suggestions for existing items that can be added, so we can give them access to that as well.

**Comment by Gandalf2k15** [comment]
Rebump to 45.

**Comment by Kroto-Ker** [comment]
Updating the TC count to such a high number is going to have a lot of negative consequences if not handled correctly. 30 was fine without a large overhaul of prices, but now that there's nukie gear and more TC- I'd say the entire pricing system needs to be re balanced for it to be functional. Personally I don't know how I feel about buffing traitors, as Security themselves weren't really.., /Buffed/. And the whole 'arms race' deal, like dragon ball or something, one guy gets stronger, the other guy gets stronger, infinite loop until the stakes are so high it destroys the station. 

Basically- I think almost ALL TC prices need to be re balanced in order for this to work, somewhat like how Baystation does it. They have 100+ TC- But some of the prices are inflated as well. They can still order more then a typical TG traitor, but it really depends on the item. You can get rifle and armor with enough left over for a few small items, or you can get a shit ton of Emags so you /always/ have one on you. Maybe that's planned in the future, but I think it needs to be done now, if you're going to change the TC values now.

**Comment by MLGTASTICa** [comment]
> Updating the TC count to such a high number is going to have a lot of negative consequences if not handled correctly. 30 was fine without a large overhaul of prices, but now that there's nukie gear and more TC- I'd say the entire pricing system needs to be re balanced for it to be functional. Personally I don't know how I feel about buffing traitors, as Security themselves weren't really.., /Buffed/. And the whole 'arms race' deal, like dragon ball or something, one guy gets stronger, the other guy gets stronger, infinite loop until the stakes are so high it destroys the station.
> 
> Basically- I think almost ALL TC prices need to be re balanced in order for this to work, somewhat like how Baystation does it. They have 100+ TC- But some of the prices are inflated as well. They can still order more then a typical TG traitor, but it really depends on the item. You can get rifle and armor with enough left over for a few small items, or you can get a shit ton of Emags so you /always/ have one on you. Maybe that's planned in the future, but I think it needs to be done now, if you're going to change the TC values now.

Security is super-buffed now, but some items could use rebalancing , like martial arts + gloves of the northstar combo.

**Comment by BrotherHangyul** [comment]
Can we keep the Katana, no matter what happens? I like it, it's good for gimmicks.

**Review by Swiftfeather — CHANGES_REQUESTED** [review]
I don't know how to use github woops

File: modular_skyrat/master_files/code/modules/uplink/uplink_items.dm [hunk 3]
```diff
@@ -87,6 +200,28 @@
...
 	exclude_modes = list(/datum/game_mode/nuclear, /datum/game_mode/nuclear/clown_ops) //It's exactly same as the one that costs 8 TC for nukies, so they have no reason to buy it for more.
 
+/datum/uplink_item/suits/standard_armor
+	name = "Standard Armor Vest"
+	desc = "A slim Type I armored vest that provides decent protection against most types of damage."
+	item = /obj/item/clothing/suit/armor/vest
+	cost = 1
+	exclude_modes = list(/datum/game_mode/nuclear, /datum/game_mode/nuclear/clown_ops) //It's exactly same as the one that costs 8 TC for nukies, so they have no reason to buy it for more.
+
+/datum/uplink_item/suits/standard_armor
```
Comment (Swiftfeather) [inline modular_skyrat/master_files/code/modules/uplink/uplink_items.dm:210]: Same uplink item path as standard armor

**Comment by YakumoChen** [comment]
I have to say this is getting to the point where it is legitimately ridiculous. We just nerfed a bunch of highball traitor items like the power sink just last week, only to make them fairly affordable in a standard kit once again.

File: modular_skyrat/master_files/code/modules/uplink/uplink_items.dm (hunk 3 above)
Comment (Gandalf2k15) [inline modular_skyrat/master_files/code/modules/uplink/uplink_items.dm]:
```suggestion
/datum/uplink_item/suits/standard_armor_traitor
```

**Comment by Gandalf2k15** [comment]
Full merging this so we don't get a build up of conflicts.
